### PR TITLE
support for kovan

### PR DESCRIPTION
**Pull request by dkent600**
Resolves: https://github.com/daostack/alchemy/issues/1558

* doesn't support any of the genericscheme contracts

### CODE REVIEW

**Comment by dkent600** [comment]
Blocked, waiting for upgrades to migration and client

File: src/components/Scheme/SchemeOpenBountyPage.tsx [hunk 1]
```diff
@@ -41,9 +41,8 @@ export default class SchemeOpenBounty extends React.Component<IProps, IState> {
   private getApi = async (): Promise<void> => {
     // query all open bounties issued by current dao ordered by decending price
     // rinkeby api and mainnet api are different subdomains
-
     const network = (await getNetworkName()).toLowerCase();
-    const testnet = network === "rinkeby" ? "rinkeby." : "";
+    const testnet = network === "main" ? "" : `${network}.`;
```
Comment (dkent600) [inline src/components/Scheme/SchemeOpenBountyPage.tsx:45]: will this work for xDai?

**Comment by orenyodfat** [comment]
@dkent600  https://alchemy-supportkovan-nuejtqnov.herokuapp.com/dao/0x7b32d2d56f32ce6523a4dee9008ccec2619a6122

File: src/arc.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import { NotificationStatus } from "reducers/notifications";
-import { getNetworkId, getNetworkName } from "./lib/util";
+import { getNetworkId, getNetworkName, targetedNetwork } from "./lib/util";
```
Comment (orenyodfat) [inline src/arc.ts:2]: why this change ?

File: src/arc.ts [hunk 3]
```diff
@@ -19,37 +19,6 @@ let selectedProvider: any;
 let web3Modal: Web3Modal;
 let initializedAccount: Address;
 
-export type Networks = "main"|"rinkeby"|"ganache"|"xdai"
```
Comment (orenyodfat) [inline src/arc.ts:22]: why this is removed ?

File: src/lib/util.ts [hunk 4]
```diff
@@ -268,12 +299,12 @@ export async function getNetworkName(id?: string): Promise<string> {
     case "main":
     case "1":
       return "main";
-    case "morden":
-    case "2":
-      return "morden";
-    case "ropsten":
-    case "3":
-      return "ropsten";
+    // case "morden":
```
Comment (orenyodfat) [inline src/lib/util.ts:302]: where is "kovan" ?

File: src/settings.ts [hunk 5]
```diff
@@ -66,6 +66,46 @@ function getWeb3ConnectProviderOptions(network: string) {
...
+      return {
+        network: "kovan",
+        torus: {
+          package: Torus,
+          options: { network: "kovan" },
+        },
+        walletconnect: {
+          package: isMobileBrowser() ? null : WalletConnectProvider,
+          options: {
+            infuraId: "e0cdf3bfda9b468fa908aa6ab03d5ba2",
```
Comment (orenyodfat) [inline src/settings.ts:79]: does this id is the same for all networks ?

File: src/arc.ts (hunk 2 above)
Comment (dkent600) [reply src/arc.ts:2]: see the corresponding change and the comment where I explained, in lib/util.ts

File: src/lib/util.ts (hunk 4 above)
Comment (dkent600) [reply src/lib/util.ts:302]: in the code, line 315

File: src/arc.ts (hunk 2 above)
Comment (orenyodfat) [reply src/arc.ts:2]: is that a must for supporting kovan ?

File: src/settings.ts (hunk 5 above)
Comment (dkent600) [reply src/settings.ts:79]: I am assuming so, am planning to test the assumption.  We should of course test all of the wallets.

File: src/arc.ts (hunk 3 above)
Comment (dkent600) [reply src/arc.ts:22]: see the corresponding change and the comment where I explained in the code, line 136 in lib/util.ts

File: data/tokens.json [hunk 6]
```diff
@@ -91,6 +91,37 @@
...
+      "0x9f13dbe2ef9733a4bb580412e9cb6f463e9bdf1a": {
+        "decimals": 18,
+        "name": "KyberNetwork",
+        "symbol": "KNC"
+      },
+      "0x9be1001d601102ae0f24ab4764dd5ce2f3e5b096": {
+        "decimals": 6,
+        "name": "USD Coin",
+        "symbol": "USDC"
+      }
```
Comment (orenyodfat) [inline data/tokens.json:122]: please exclude all these tokens from kovan . 
it is not exist or the addresses are different at kovan

File: src/arc.ts (hunk 2 above)
Comment (dkent600) [reply src/arc.ts:2]: Without that change we would be worsening the problem by adding a dependency to the circle. 
 There is less risk in fixing this than in not (which is easy to say because this change is virtually risk-free).

File: src/components/Scheme/SchemeOpenBountyPage.tsx (hunk 1 above)
Comment (orenyodfat) [reply src/components/Scheme/SchemeOpenBountyPage.tsx:45]: you are asking if we might have the bounty scheme in xdai ?
so ,might be though not now ..and if yes so any how the scheme json will need to be updated.
so I suggest to remove this update.

File: src/arc.ts (hunk 3 above)
Comment (orenyodfat) [reply src/arc.ts:22]: is this change is a must to support kovan or any other network ?

File: src/components/Scheme/SchemeOpenBountyPage.tsx (hunk 1 above)
Comment (dkent600) [reply src/components/Scheme/SchemeOpenBountyPage.tsx:45]: is bounty scheme in kovan?

File: src/settings.ts (hunk 5 above)
Comment (orenyodfat) [reply src/settings.ts:79]: why this infura id is in the code at all ? it should come as env var . it is private info , no ?

File: src/lib/util.ts (hunk 4 above)
Comment (orenyodfat) [reply src/lib/util.ts:302]: why there is rinkeby support in this function and not kovan ?

File: src/settings.ts (hunk 5 above)
Comment (dkent600) [reply src/settings.ts:79]: I'm not sure all the wallets are going to work.

File: src/components/Scheme/SchemeOpenBountyPage.tsx (hunk 1 above)
Comment (orenyodfat) [reply src/components/Scheme/SchemeOpenBountyPage.tsx:45]: not yet. and if there will be it will require updating the json for this scheme. 
so again , please remove this update

File: src/settings.ts (hunk 5 above)
Comment (orenyodfat) [reply src/settings.ts:79]: why we have the infura token spread all over the code ?
it should get into env var . see #1569

File: src/components/Scheme/SchemeOpenBountyPage.tsx (hunk 1 above)
Comment (orenyodfat) [reply src/components/Scheme/SchemeOpenBountyPage.tsx:45]: leaving this update will require us to verify that the rinkeby bounty scheme is working as before with this change . did you ?

File: src/lib/util.ts (hunk 4 above)
Comment (dkent600) [reply src/lib/util.ts:302]: the support for kovan is there, please check the code

File: src/arc.ts (hunk 3 above)
Comment (dkent600) [reply src/arc.ts:22]: same answer as above because this is part of the same change.

Without that change we would be worsening the problem by adding a dependency to the circle.  There is less risk in fixing this than in not (which is easy to say because this change is virtually risk-free).

It is not a must for this PR, unless you're worried about the potential for webpack to behave badly because of circular module dependencies.

File: src/settings.ts (hunk 5 above)
Comment (dkent600) [reply src/settings.ts:79]: that is for another PR

File: data/tokens.json (hunk 6 above)
Comment (dkent600) [reply data/tokens.json]: done

File: src/lib/util.ts (hunk 4 above)
Comment (orenyodfat) [reply src/lib/util.ts:302]: at the getNetworkName ?

File: src/components/Scheme/SchemeOpenBountyPage.tsx (hunk 1 above)
Comment (dkent600) [reply src/components/Scheme/SchemeOpenBountyPage.tsx:45]: The logic is so simple, and the way to test it not trivial (in fact not possible since we can't use rinkeby), that I haven't worried about it.

Comment (dkent600) [reply src/components/Scheme/SchemeOpenBountyPage.tsx:45]: Maybe @EzraWeller can test viewing bounties when he gets to this.

File: src/lib/util.ts (hunk 4 above)
Comment (dkent600) [reply src/lib/util.ts:302]: Here: https://github.com/daostack/alchemy/blob/a84a2ae762e869cc3a107644f3e5bd35f9a16335/src/lib/util.ts#L314
